vllm/model_executor/models/deepseek_v2.py

```diff
@@ -394,33 +398,56 @@ def __init__(
         self.padding_idx = config.pad_token_id
         self.vocab_size = config.vocab_size
 
-        self.embed_tokens = VocabParallelEmbedding(
-            config.vocab_size,
-            config.hidden_size,
+        if get_pp_group().is_first_rank:
+            self.embed_tokens = VocabParallelEmbedding(
+                config.vocab_size,
+                config.hidden_size,
+            )
+        else:
+            self.embed_tokens = PPMissingLayer()
+
+        self.start_layer, self.end_layer, self.layers = make_layers(
+            config.num_hidden_layers,
+            # layer_idx is still an argument
+            functools.partial(DeepseekV2DecoderLayer,
+                              config,
+                              cache_config=cache_config,
+                              quant_config=quant_config),
```

Comment:
this lambda function will have `prefix=` shortly after https://github.com/vllm-project/vllm/pull/6515 .